packages/toolpad-app/src/appDom/index.ts

```diff
@@ -727,14 +727,34 @@ export function moveNode<Parent extends AppDomNode, Child extends AppDomNode>(
   return setNodeParent(dom, node, parent.id, parentProp, parentIndex);
 }
 
-export function duplicateNode<Child extends ElementNode>(dom: AppDom, node: Child) {
+export function duplicateNode<Parent extends AppDomNode, Child extends ElementNode>(
+  dom: AppDom,
+  node: Child,
+  parent?: Parent,
+) {
   if (!node.parentId || !node.parentProp || !node.parentIndex) {
     throw new Error(`Node: "${node.id}" can't be duplicated`);
   }
 
+  const { children } = getChildNodes(dom, node);
+
   const newNode = createElement(dom, node.attributes.component!.value, node.props, node.layout);
 
-  return setNodeParent(dom, newNode, node.parentId, node.parentProp, node.parentIndex);
+  let updatedDom = dom;
+
+  if (Array.isArray(children)) {
```

Comment:
Which values could `children` have? If it could be an array or null/undefined maybe we could just check for if the value is truthy or falsy?